### PR TITLE
Feature | Routes and bin:lib crate architecture

### PR DESCRIPTION
**Pull request by EstebanBorai**
In order to provide a more HTTP server idiomatic structure
a "router" module is implemented in charge of "wiring up"
HTTP services to the main Actix instance.

The structure of the project is reworked as well to accomplish
a the Rust convention of disengaging executable logic from
application logic.

### CODE REVIEW

**Comment by EstebanBorai** [comment]
@morenol would be nice to have #2 merged before going through this PR.

Anyway if you wanna take a look go ahead! Let me know any ideas/suggestions please!

File: Cargo.toml
Comment (EstebanBorai) [inline Cargo.toml:2]: Improves simplicity for `use` statements.

So instead of having to do:

```rust
use hilow_social_feed_api;
```

We would do instead:

```rust
use hilow;
```

File: src/bin/main.rs
Comment (EstebanBorai) [inline src/bin/main.rs:22]: I suggest to keep `main.rs` as clean as possible, just resources initializations and Actix startup configuration allowed.

This would help us to have a clean project structure, I really want to encourage the use of a router in a separated module to avoid importing `handlers` and any domain related values to the HTTP server initialization.

What do you guys think?

File: src/handlers/hello.rs
Comment (EstebanBorai) [inline src/handlers/hello.rs:5]: This is temporal, and must go away when we begin to define handlers/endpoints, its just an example and it also helps us having a valid module to import

File: src/router.rs
Comment (EstebanBorai) [inline src/router.rs:8]: `router` is in charge of assigning a `handler` to a `resource` (or path).
This structure will help us to prefix multiple handlers on a DRY manner, having a router defined like these also help us navigating through the project and reaching to endpoints implementations with ease.

**Review by rafacouto — COMMENTED** [review]
Good job!

File: src/bin/main.rs
Comment (morenol) [inline src/bin/main.rs:1]: I have a doubt here: doesn't this change require changes to the Cargo.toml file?
I have seen this in the Cargo.toml file for binaries. I guess that for the main binary it is not necessary, Am I right?
```toml
[[bin]]
name = "main"
test = false
doctest = true
bench = false
```

**Review by morenol — APPROVED** [review]
LGTM, I am Ok with merging this once #2 is merged

File: src/bin/main.rs
Comment (EstebanBorai) [reply src/bin/main.rs:1]: The Cargo.toml configuration for binary crates is supported OOTB, when you use `main.rs` and `lib.rs`  files in your `src` directory. `cargo` knows how to execute this crate when you execute it.

Here is an I did when reading the book, building the `minigrep` solution:

https://www.github.com/EstebanBorai/the-rust-programming-language/tree/main/minigrep%2FCargo.toml

It's called "Separation of Concerns for Binary Projects".

When you want to have something else which doesn't follows this convention, then you can always overwrite this behavior using the configuration you provided above.

**Review by rafacouto — APPROVED** [review]
LGTM

**Pull request merged**
